src/core/browser/qgsfilebaseddataitemprovider.h

```diff
@@ -0,0 +1,131 @@
+/***************************************************************************
+  qgsfilebaseddataitemprovider.h
+  --------------------------------------
+  Date                 : July 2021
+  Copyright            : (C) 2021 by Nyall Dawson
+  Email                : nyall dot dawson at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSFILEBASEDDATAITEMPROVIDER_H
+#define QGSFILEBASEDDATAITEMPROVIDER_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgsdataitemprovider.h"
+#include "qgsdatacollectionitem.h"
+#include "qgslayeritem.h"
+#include "qgsprovidersublayerdetails.h"
+#include <QString>
+#include <QVector>
+
+class QgsProviderSublayerDetails;
+
+#define SIP_NO_FILE
+
+class QgsDataItem;
+
+/**
+ * \ingroup core
+ * \brief A generic data item for file based layers.
+ *
+ * This is a generic data item for file based layers. It is created by a QgsFileBasedDataItemProvider
+ * for files which represent a single layer, or as children of a QgsFileDataCollectionItem for
+ * files which contain multiple layers.
+ *
+ * \since QGIS 3.22
+ */
+class CORE_EXPORT QgsProviderSublayerItem final: public QgsLayerItem
+{
+    Q_OBJECT
+  public:
+
+    /**
+     * Constructor for QgsProviderSublayerItem.
+     * \param parent parent item
+     * \param name data item name (this should match either the layer's name or the filename of a single-layer file)
+     * \param details sublayer details
+     * \param isFile set to TRUE if the item represents a single-layer file, or FALSE if the item is a sublayer from a
+     * collection item.
+     */
+    QgsProviderSublayerItem( QgsDataItem *parent, const QString &name, const QgsProviderSublayerDetails &details, bool isFile );
```

Comment:
I'm not really sure but I think I'd prefer an `isSubLayer` flag instead of `isFile`.

Reply:
Can i leave as is for the moment? I'm wanting to add some type of flag to QgsDataItem later to reflect that the item is a file, and use this in the gui providers to ensure that the file related operations (such as "file properties") are always shown for these items, regardless of their subclass or provider.